chain/block.go

```diff
@@ -890,10 +892,78 @@ func (b *StatefulBlock) Results() []*Result {
 	return b.results
 }
 
-func (b *StatefulBlock) FeeManager() *fees.Manager {
+func (b *StatefulBlock) FeeManager() *internalfees.Manager {
 	return b.feeManager
 }
 
+type ExecutedBlock struct {
+	Block      *StatelessBlock `json:"block"`
```

Comment:
I would just prefer if we manually copied out the fields in the `StatelessBlock` type (Parent, Timestamp, Height, Txs, StateRoot). `StatelessBlock` has weird signatures (e.g `Size` is a redundant method because you can call `len` on the result of `Marshal`, and the call to `ID()` returns an error but we could handle that more cleanly by just initializing the `ExecutedBlock` with an `ID`) that I want to avoid propagating to upstream code.